src/arrayMethodJoin.js

```diff
@@ -4,8 +4,36 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    let string = '';
+
+    for (let i = 0; i < this.length; i++) {
+      switch (this[i]) {
+        case this[this.length - 1]:
+          switch (this[this.length - 1]) {
+            case null:
+            case undefined:
+              string += '';
+              break;
+
+            default:
+              string += this[this.length - 1];
+              break;
+          };
+          break;
+
+        case null:
+        case undefined:
+          string += '' + `${separator}`;
+          break;
+
+        default:
+          string += this[i] + `${separator}`;
+          break;
+      }
```

Comment:
These checks are unnecessary, they degrade performance. It will be better to do it according to the following algorithm:

Each time add text and a separator to the resulting string, and at the end already check for the last element and add only the string without the separator.

Reply:
done